Automattic/jetpack
# [RNMobile] Fix - Allow uploads to finish outside the editor by setting the local media ID

Related to https://github.com/wordpress-mobile/gutenberg-mobile/issues/5570 and https://github.com/wordpress-mobile/gutenberg-mobile/issues/5571.

## Proposed changes:

* The app can finish video uploads using upload processors ([example](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergVideoUploadProcessor.swift)), these processors update the post content after the upload with the proper attributes and content. However, in order to associate the media items being uploaded with blocks to be updated, we need to set the local media ID as a reference.
* When the upload starts, we set the media local id as the `id` attribute.

### Other information:


## Jetpack product discussion


N/A

## Does this pull request change what data or activity we track or use?


N/A

## Testing instructions:






This change relies on the upload processors from the app so it's needed to test with the following PRs:
- **iOS:** https://github.com/wordpress-mobile/WordPress-iOS/pull/20380
- **Android:** 🚧 

1. Create/open a post.
2. Add a VideoPress block.
3. Select the "Choose from device" option.
4. Select a video.
5. While there’s an ongoing upload, close the post with publishing changes.
6. Verify that you see the upload progress in the post summary.
7. Re-open the post.
8. Preview the post and observe that the video can be played.
